### ISSUE TITLE
Combat Realism issue - Ammo in inventory getting destroyed causes explosion at location it was picked up

### ISSUE THREAD

**Issue by Jalordon** [skyarkhangel/Hardcore-SK#206]
Explosions that cause enough damage to a pawns inventory ammo will cause an explosion at where the ammo was picked up. This is really bad when the ammo was in a large stock pile as it chain reacts and blows up the entire ammo storage when the ammo that exploded was no where near the stockpile.

In practical game play, I noticed this issue happening when mechanoid inferno cannons hit my pawns, though playing around in dev mode I got the same to happen with incendiary grenades.

Steps to reproduce (Doesn't happen always, may take a few attempts if the ammo in inventory does not get destroyed before the pawn is incapacitated. If they get incapacitated before the ammo is destroyed, they will drop the ammo at where they are and it will explode where it is and not where it was picked up):
1. Spawn a bunch of stacks of ammo adjacent to eachother (2x2 or 3x3 should be more than enough to see it) that can explode such as 9mm far enough away that it is certain that the explosion used later does not hit the ammo stockpile.
2. Pick up a stack of ammo with a pawn
3. With a second pawn, give it a mechanoid inferno cannon or incendiary grenade
4. Shoot explosives at the pawn with ammo in its inventory
5. Notice that the ammo stockpile explodes despite being well away from the pawn that was in an explosion.

edit: Secondary issue that may need its own report. Ammo that gets damaged to play the cook off (sparking on fire animation) never actually explodes until it hits 0 health. This lasts even if the ammo is picked up and dropped, and also if the ammo is repaired at a mending table.

**Comment by jnkyacc** [comment]
I think CR consider inventory stack and stockpile stack are the same(id) until you drop inventory stack.

**Comment by Jalordon** [comment]
I noticed in the save file that the saved inventory stores a position for each apparel and inventory item, which I took as a guess that the item is technically at the spot it was picked up to reduce processing needed by not updating the position. But this is just me making guesses. I think that's what you said but just reiterating.

No idea what functions are available but it would be good to update position on being damaged/destroyed. This is just me saying stuff not knowing what functions are available to use.
